nixos/modules/services/logging/graylog.nix

```diff
@@ -35,8 +35,8 @@ in
 
       package = lib.mkOption {
         type = lib.types.package;
-        default = if lib.versionOlder config.system.stateVersion "23.05" then pkgs.graylog-3_3 else pkgs.graylog-5_1;
-        defaultText = lib.literalExpression (if lib.versionOlder config.system.stateVersion "23.05" then "pkgs.graylog-3_3" else "pkgs.graylog-5_1");
+        default = pkgs.graylog-6_0;
+        defaultText = lib.literalExpression "pkgs.graylog-6_0";
```

Comment:
I'm not so happy with silently changing the default for all state versions. It's probably the easiest to add an assertion for old state versions like in https://github.com/NixOS/nixpkgs/pull/345625/files#diff-5834a9cebfb2c9d2d6863e68be9c225f325f77d1c5a1a5e55f66fa3298995a93R90 or only set the default when the state version is >=25.05

Reply:
Thank you very much for pointing out the assertion pattern that I was not aware of.

However, after grep'ing for this pattern and recognizing that it is only used for UniFi, I decided to go with the pattern PostgreSQL uses:

in the options part
https://github.com/NixOS/nixpkgs/blob/924dd674814aaf618e329dd252e07f9a083c5f20/nixos/modules/services/databases/postgresql.nix#L89-L91

In the implementation part (`config = mkIf cfg.enable`)
https://github.com/NixOS/nixpkgs/blob/924dd674814aaf618e329dd252e07f9a083c5f20/nixos/modules/services/databases/postgresql.nix#L491-L512